Base/QTCore/qSlicerExtensionsManagerModel.cxx

```diff
@@ -802,6 +811,138 @@ void qSlicerExtensionsManagerModelPrivate::saveExtensionDescription(
+}
+
+// --------------------------------------------------------------------------
+void qSlicerExtensionsManagerModelPrivate::scheduleExtensionHistorySettingRemoval(
+	const QString& extensionsHistorySettingsFile, const ExtensionMetadataType &extensionMetadata)
+{
+	this->addExtensionHistorySetting(extensionsHistorySettingsFile, extensionMetadata, "ExtensionsHistory/ScheduledForRemoval");
+}
+
+// --------------------------------------------------------------------------
+void qSlicerExtensionsManagerModelPrivate::addExtensionHistorySetting(
+	const QString& extensionsHistorySettingsFile, const ExtensionMetadataType &extensionMetadata, const QString& settingsPath)
+{
+	QSettings settings(extensionsHistorySettingsFile, QSettings::IniFormat);
+	QStringList& settingsInfoList = settings.value(settingsPath).toStringList();
+  settingsInfoList << extensionMetadata.value("extensionname").toString();
+	settingsInfoList.removeDuplicates();
+	settings.setValue(settingsPath, settingsInfoList);
+}
+
+// --------------------------------------------------------------------------
+void qSlicerExtensionsManagerModelPrivate::cancelExtensionHistorySettingRemoval(
+	const QString& extensionsHistorySettingsFile, const QString& extensionName)
+{
+	QSettings settings(extensionsHistorySettingsFile, QSettings::IniFormat);
+	QStringList& settingsInfoList = settings.value("ExtensionsHistory/ScheduledForRemoval").toStringList();
+  settingsInfoList.removeOne(extensionName);
+  settingsInfoList.removeDuplicates();
+  settings.setValue("ExtensionsHistory/ScheduledForRemoval", settingsInfoList);
+}
+
+// --------------------------------------------------------------------------
+void qSlicerExtensionsManagerModelPrivate::removeScheduledExtensionHistorySettings(
+	const QString& extensionsHistorySettingsFile)
+{
+	QSettings settings(extensionsHistorySettingsFile, QSettings::IniFormat);
+	QStringList& scheduledForRemovalList = settings.value("ExtensionsHistory/ScheduledForRemoval").toStringList();
+	QStringList& historyList = settings.value("ExtensionsHistory/Revisions/" + this->SlicerRevision).toStringList();
+	for (unsigned int i = 0; i < scheduledForRemovalList.length(); i++)
+	{
+		historyList.removeOne(scheduledForRemovalList.at(i));
+	}
+	historyList.removeDuplicates();
+	settings.setValue("ExtensionsHistory/Revisions/" + this->SlicerRevision, historyList);
+	settings.setValue("ExtensionsHistory/ScheduledForRemoval", "");
+}
+
+// --------------------------------------------------------------------------
+QVariantMap qSlicerExtensionsManagerModelPrivate::getExtensionsInfoFromPreviousInstallations(
+	const QString& extensionsHistorySettingsFile)
+{
+	Q_Q(qSlicerExtensionsManagerModel);
+	QVariantMap extensionsHistoryInformation;
+	QSettings settings(extensionsHistorySettingsFile, QSettings::IniFormat);
+	settings.beginGroup("ExtensionsHistory/Revisions");
+	QStringList& revisions = settings.childKeys();
+	revisions.sort();	//revisions sorted ascending
```

Comment:
It should be a numeric sort instead of a string sort. Also, it may be confused when there are privately built Slicer versions that use git tags for version information.

We'll have to keep this in mind for git migration (as tags are not sequential). Maybe add a comment that contains SVN and git words so that we can easily find this code and update it to work with git.